### ISSUE TITLE
Android background

### ISSUE THREAD

**Issue by VyacheslavMartynenko** [square/okhttp#4107]
Per documentation, android Foreground Services are not affected by doze mode. But even you use socket client inside such service when you press the power button and turn off the charging, you will get SocketTimeoutException because pong is not received on time. Can we create a stable socket connection in any device condition using this library?

**Comment by swankjesse** [comment]
Can you rephrase this as a request?

**Comment by VyacheslavMartynenko** [comment]
To be honest, I don't know which category is better to relate this issue. I've seen other issues before, where the implementation was in the background service. And in the response, it was said that the connection is lost because of the platform limitations, and you need to use the foreground service to circumvent these restrictions. But after trying such an implementation, the result remained the same. And I would like to understand whether it is possible to create a stable socket connection in any device condition?

**Comment by VyacheslavMartynenko** [comment]
Yeah, I understand this. I asked a question on StackOverflow, but I did not get any answers to it. So, although the question doesn't directly apply to the library, I decided to clarify this point here, because, at the time of development, you may have received an answer to this question. I apologize for the inconvenience, if delivered them.

**Comment by swankjesse** [comment]
Link to the StackOverflow issue?

**Comment by yschimke** [comment]
Which version of Android is this on for example?

**Comment by VyacheslavMartynenko** [comment]
[https://stackoverflow.com/questions/51133783/socket-connection-in-background](url)

**Comment by swankjesse** [comment]
I think this is more of an Android question than an OkHttp question. I'm interested to find out if there is a way to keep the web socket open, but I don't think theres any action we can take here.

**Comment by VyacheslavMartynenko** [comment]
Thank you for your response.

**Comment by VyacheslavMartynenko** [comment]
So I finished my research, and I can say that the problem belongs to Android. The ways to solve it can be seen in the answer to the StackOverflow on the same link.

Best regards,
Vyacheslav
